packages/snowpack/src/build/build-pipeline.ts

```diff
@@ -54,11 +54,17 @@ async function runPipelineLoadStep(
         });
       },
     });
-    const mainOutputExt = step.resolve.output[0];
     if (typeof result === 'string') {
+      const mainOutputExt = step.resolve.output[0];
       return {[mainOutputExt]: result};
     } else if (result && typeof result === 'object') {
-      return result;
+      const output: Record<string, string> = {};
+      Object.entries(result).forEach(([ext, code]) => {
```

Comment:
nit: should we throw an error here instead? If the plugin dev isn't being good about using a `.` here, they may be forgetting it somewhere else (ex: in their fileExt check).